src/test/e2e-emulated/contracts/semantics.tact

```diff
@@ -40,6 +40,95 @@ extends mutates fun changeSomeFields(self: SA) {
     self.a2.b2.c1 += 1;
 }
 
+fun getNewMap(m: map<Int,SA>): map<Int,SA> {
+    // Reassign the parameter value
+    // This does not change the map passed as parameter to the function
+    m = emptyMap();
+    
+    m.set(1, SA {a1: 0, a2: SB {b1: false, b2: SC {c1: 0}, b3: 0}});
+
+    return m;
+}
+
+fun copyAndModifyMap(m: map<Int,SA>): map<Int,SA> {
+    // Change some entries in the map given as parameter
+    // Since maps are passed by value, this does not change the map passed as parameter to the function
+
+    // Note that it is not possible to directly change the struct in entry 1 like 
+    // m.get(1)!!.a2.b2.c1 += 40;
+    // because the compiler only allows path expressions on the left of assignments.
+    // So, we need to read m.get(1) and store it in a variable:
+
+    let c = m.get(1)!!;
+
+    // Then change the struct field here
+    c.a2.b2.c1 += 40;
+
+    // And then reassign entry 1
+    // If we do not reassign entry 1 in m, the change we did in the previous line
+    // to c will not be reflected in the map m, because struct c is a copy!!! (Recall that
+    // structs are assigned by value as well).
+    // To check the previous claim, this function will return an emptyMap()
+    // if m.get(1)!!.a2.b2.c1 == c.a2.b2.c1 at this point (which will not be true):
+
+    if (m.get(1)!!.a2.b2.c1 == c.a2.b2.c1) {
+        return emptyMap();
+    }
+
+    m.set(1, c);
+
+    // Now delete entries 2 and 3
+
+    m.del(2);
+    m.del(3);
+
+    return m;
+}
+
+/*
+The following two functions are not allowed by the compiler, because mutating functions 
```

Comment:
Let's open an issue on this, looks like an important language feature to have in Tact. Btw, a workaround for this would be to wrap a map into a struct and then it's suddenly allowed :)

Reply:
Ohhh, good point. I never thought of putting a map inside a struct because I tested structs first. But now I see that I need to add tests that combine things that I previously tested.